src/mds/MDSRank.cc

```diff
@@ -1919,20 +1919,30 @@ bool MDSRankDispatcher::handle_asok_command(
   } else if (command == "dump cache") {
     Mutex::Locker l(mds_lock);
     string path;
+    int r;
     if(!cmd_getval(g_ceph_context, cmdmap, "path", path)) {
-      mdcache->dump_cache(f);
+      r = mdcache->dump_cache(f);
     } else {
-      mdcache->dump_cache(path);
+      r = mdcache->dump_cache(path);
+    }
+
+    if (r != 0) {
+      ss << "Failed to dump cache: " << cpp_strerror(errno);
     }
   } else if (command == "dump tree") {
     string root;
     int64_t depth;
+    int r;
     cmd_getval(g_ceph_context, cmdmap, "root", root);
     if (!cmd_getval(g_ceph_context, cmdmap, "depth", depth))
       depth = -1;
     {
       Mutex::Locker l(mds_lock);
-      mdcache->dump_cache(root, depth, f);
+      r = mdcache->dump_cache(root, depth, f);
```

Comment:
if you move the `ss <<...` bit into this block, then you could do `int r = mdcache->dump_cache...` and thereby restrict the scope of the `r` variable.